### DISCUSSION TITLE
Lottie animations on page transition

### DISCUSSION THREAD

**Discussion by andresclua** [swup/swup#786] [Help] [answered]
I all!  -  hope you're doing well!

I'm currently working on a project that utilizes Swup 4 & Astro.

However, I've encountered an issue with Lotties (I'm using this package - https://www.npmjs.com/package/@lottiefiles/lottie-player).

When I load each page independently, everything works perfectly without any problems. However, I've added a script inside my layout, and it looks something like this:

```
<html>
<head>
<script src="https://unpkg.com/@lottiefiles/lottie-player@1.5.7/dist/lottie-player.js"></script>
</head>
<body>
  <main id="swup" class="wrapper">
            <slot />
        </main>
</body>

```

Any tip on how to solve this?

![Screenshot 2023-09-18 at 15 24 50](https://github.com/swup/swup/assets/612806/e1ce15e2-90d8-4da4-9b52-b62ea438fd34)

This error shows up on page transition.

Would you need a sandbox to debug it? my project is on the finishline, so would love to get help ASAP !

Thanks in advance! 

Cc: @nereafontecha

**Comment by daun** [comment]
Hey @andresclua — What seems to be working well for Astro projects in these scenarios is installing the [Head Plugin](https://swup.js.org/plugins/head-plugin/) as well as the [Scripts Plugin](https://swup.js.org/plugins/scripts-plugin/). Personally, I'd be wary of raw script tags in general — usually with Astro the cleanest solution is an import from npm in a hydrated client-side component. If the combination of the two plugins doesn't help, we'll probably need a sandbox to take a proper look.

**Reply by andresclua** [reply, marked as answer]
100% it was that! - 😓

**Reply by daun** [reply]
What helped? The plugins or switching to an npm import?

**Reply by andresclua** [reply]
It was [this](https://swup.js.org/plugins/scripts-plugin/) plugin !

Added data-swup-ignore-script and that's it. The minute I was reading your reply saw this [Head Plugin](https://swup.js.org/plugins/head-plugin/) I said, "nooooo" LOL.

Thanks for your time & help @daun

**Reply by daun** [reply]
Haha, glad you got it working!
